frontend/packages/dev-console/src/components/catalog/utils/custom-catalog-categories.ts

```diff
@@ -0,0 +1,47 @@
+import { CatalogCategory, CatalogSubcategory, CatalogCategories } from './types';
+
+export interface DeveloperConsoleCatalogCategory {
+  id: string;
+  label: string;
+  tags?: string[];
+  subcategories?: DeveloperConsoleCatalogSubcategory[];
+}
+
+export interface DeveloperConsoleCatalogSubcategory {
+  id: string;
+  label: string;
+  tags?: string[];
+}
+
+const mapSubcategory = (subcategory: DeveloperConsoleCatalogSubcategory): CatalogSubcategory => ({
+  id: subcategory.id,
+  label: subcategory.label,
+  field: 'tags',
+  values: subcategory.tags,
+});
+
+const mapSubcategories = (
+  subcategories: DeveloperConsoleCatalogSubcategory[],
+): Record<string, CatalogSubcategory> => {
+  const out: Record<string, CatalogSubcategory> = {};
+  subcategories.forEach((subcategory) => {
+    out[subcategory.id] = mapSubcategory(subcategory);
+  });
+  return out;
+};
+
+const mapCategory = (category: DeveloperConsoleCatalogCategory): CatalogCategory => ({
+  id: category.id,
+  label: category.label,
+  field: 'tags',
+  values: category.tags,
```

Comment:
We should align the catalog apis and the console config schema for catalog categories.
We should be able to drop `field` and change `values` to `tags`.

Reply:
@christianvogt @rohitkrai03 I changed the default categories now to the CRD format. (Arrays instead of objects and replaced field and values with tags. Changed all necessary frontend code to keep the arrays up to rendering.